public/javascript/main.js

```diff
@@ -376,5 +376,62 @@ $(function(){
 
           $(this).data('chart', chart);
       }
-  })
+  });
+
+  $(".js-download-with-term").on("click", function(e){
+    var term = $(e.target).data("term");
+    var house = $(e.target).data("house");
+    if (term && house) {
+        // inject query vars into URL before the target anchor
+        e.preventDefault();
+        location = $(e.target).attr('href')
+          .replace('#', "?" + $.param({term: term, house: house}) + "#");
+    }
+  });
+
+  if (typeof pageInitFunction === 'function') {
+      pageInitFunction();
+  }
+
 });
+
+// collapse_displayed_items:
+// If there are too many items being displayed in the list, hide
+// all but a few, making sure the target element (if there is one)
+// is included in those that are not hidden. Add a button for
+// revealing the hidden items.
+// Note this is used on the download page, where (currently) there
+// may be more than one such list (e.g., one for each legislature).
+
+function collapse_displayed_items(
+    $list_of_elements,   // list of elements that need to be collapsed
+    $target_element,     // item within that which needs to be highlighted
+                         // ...if none (which is OK) take the first
+    hidden_class_name,   // CSS class used to distinguish collapsed elements
+    min_threshold_items, // only collapse if more elements than this
+    max_display_items,   // number of elements to show when collapsed
+    $button_for_reveal   // button that is added to reveal
+  ) {
+  if ($list_of_elements.size() > min_threshold_items) {
```

Comment:
Use `.length` instead of `.size()`<sup>[1]</sup>

<sup>[1]</sup> https://api.jquery.com/size/ – “The .size() method is deprecated as of jQuery 1.8. Use the .length property instead.”